### PR TITLE
single precision support in skimage.filters

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR contains the commits from #5353, so that one should be reviewed prior to this (float32 support Hessian functions in skimage.feature were required in the ridge filters implementation here).

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/filters/lpi_filter.py [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
-from .._shared.utils import check_nD
+from .._shared.utils import _supported_float_type, check_nD
 
 eps = np.finfo(float).eps
 
 
 def _min_limit(x, val=eps):
-    mask = np.abs(x) < eps
-    x[mask] = np.sign(x[mask]) * eps
+    mask = np.abs(x) < val
```
Comment (rfezzani) [inline skimage/filters/lpi_filter.py:15]: :eyes: good catch :clap:

File: skimage/filters/lpi_filter.py [hunk 2]
```diff
@@ -195,7 +200,7 @@ def inverse(data, impulse_response=None, filter_params={}, max_gain=2,
         filt = predefined_filter
 
     F, G = filt._prepare(data)
-    _min_limit(F)
+    _min_limit(F, val=np.finfo(float).eps)
```
Comment (rfezzani) [inline skimage/filters/lpi_filter.py:203]:
```suggestion
    _min_limit(F, val=eps)
```
`eps` is already defined as a global variable. But I am not against your writing, in this case I suggest removing the global declaration of `eps`, just for consistency :wink:

File: skimage/filters/lpi_filter.py [hunk 3]
```diff
@@ -238,7 +243,7 @@ def wiener(data, impulse_response=None, filter_params={}, K=0.25,
...
```
Comment (rfezzani) [inline skimage/filters/lpi_filter.py:246]: Same here

**Review by rfezzani — APPROVED** [review]
Excellent, thank you @grlee77.

**Comment by grlee77** [comment]
Hi @rfezzani, thanks for reviewing this previously. I fixed conflicts and addressed your comment about the global `eps` variable.

Since the last time you reviewed this, I added single precision support to the three `threshold_*` functions that return arrays (see commits 42c9a94 and 5ebcf52).  

Also, it could go in a separate standalone PR if preferred, but there are a couple of minor fixes to prior single precision PRs in fbb1df9 (a dtype, not an image should be passed to `_supported_float_type`).

**Comment by rfezzani** [comment]
Thank you @grlee77, a second approval and we are good to go :wink: !

**Comment by grlee77** [comment]
This is the only one of the single precision PRs that has not yet been merged.  It has one approval now and the changes made here are consistent with the already merged PRs in this area. Given that, I would be inclined to merge it as is rather than ask others to spend substantial time doing additional reviews. Are there any objections to this?

**Comment by rfezzani** [comment]
> ... Are there any objections to this?

Not from my side ^^

**Pull request merged**